src/pybind/mgr/volumes/fs/subvolume.py

```diff
@@ -189,6 +189,29 @@ def get_subvolume_path(self, spec):
             raise VolumeException(-e.args[0], e.args[1])
         return path
 
+    def get_dir_names(self, path):
+        """
+        Get the directory names in a given path
+        :param path: the given path
+        :return: the list of directory names in json format
+        """
+        try:
+            dir_handle = self.fs.opendir(path)
+        except cephfs.ObjectNotFound:
+            raise VolumeException(-errno.ENOENT,
+                                  "Path '{0}' not found".format(path))
+        except cephfs.Error as e:
+            raise VolumeException(-e.args[0], e.args[1])
+
+        dirs = []
+        d = self.fs.readdir(dir_handle)
+        while d:
+            if d.d_name not in (b".", b".."):
```

Comment:
would there be a need to check if the entry is a directory?

Reply:
good catch! need this

Reply:
working on this.